src/routes/api/sections/bookSection.ts

```diff
@@ -0,0 +1,176 @@
+import { Request, Response } from 'express';
+import timeBlockModel from '../../../models/timeBlock.model';
+import sectionModel from '../../../models/section.model';
+import Room from '../../../models/room.model';
+import { DateTime } from 'luxon';
+
+const bookSectionRoute = async (req: Request, res: Response): Promise<void> => {
+    const id = req.params.id;
```

Comment:
I'd get rid of `req.params.id` and just use body arguments here - it simplifies things for the frontend by a little bit